application/bsdd_utils.py

```diff
@@ -42,4 +50,45 @@ def get_inst(instance_id):
     
     return hierarchical_bsdd_results     
 
+def bsdd_data_processing(bsdd_task, bsdd_results, session):
+    bsdd_instances = [bsdd_table(result.serialize(), session) for result in bsdd_task.results]
+    bsdd_data = defaultdict(lambda: {'valid': 0, 'invalid': 0, 'source': get_domain(bsdd_results)})
+
+    for item in bsdd_instances:
+        classification = item['classification']
+        validity = item['validity']
+        bsdd_data[classification][validity] += 1
+
+    bsdd_data = [{**{'classification': k}, **v} for k, v in bsdd_data.items()]
+    return bsdd_data
+
+def bsdd_table(bsdd_result, session):
+    inst = get_inst(session, bsdd_result['instance_id'])
+    observed_type = inst.ifc_type
+    required_type = bsdd_result['bsdd_type_constraint']
+    validity = "valid" if utils.do_try(lambda: ifcopenshell.template.create(schema_identifier="IFC4X3").create_entity(observed_type).is_a(required_type), 'invalid') else 'invalid'
+    return {'classification': observed_type, 'validity': validity}
+
+def get_inst(session, instance_id):
+    return session.query(database.ifc_instance).filter(database.ifc_instance.id == instance_id).all()[0]
+
+def get_domain(bsdd_results):
```

Comment:
That works, and I've implemented those changes in https://github.com/Ghesselink/ifc-pipeline-validation/commit/58a98f1f06ecb6a65d65104453be6857dc34b965

I understand that it's preferable to avoid duplicating the code. However, the current implementation (again invoking the get_domain function) makes a request for every classification, which significantly increases the loading time for a BSDD report. This may lead to frustration among users.

Wouldn't it be better to add this to the database at `check_bsdd_v2.py ` and retrieve at the flask route? This avoids both duplicating code as well as long loading times.